src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java

```diff
@@ -41,6 +41,15 @@ public ExceptionDetails handleException(NotFoundException notFoundException) {
         return exceptionDetailsList;
     }
 
+    @ExceptionHandler(DeletionRestrictedException.class)
+    @ResponseBody
+    @ResponseStatus(HttpStatus.FORBIDDEN)
```

Comment:
403 is used when permissions are not right. What status code to use here is tricky, but I suggest 409 (Conflict) or 412 (Precondition Failed) they suits more from my point of view

Reply:
+